packages/compiler/src/shadow_css.ts

```diff
@@ -157,6 +172,115 @@ export class ShadowCss {
     return this._insertPolyfillRulesInCssText(cssText);
   }
 
+  /*
+   * Process styles to add scope to keyframes, this implies modifying both
+   * the names of the keyframes defined in the component styles and also the css
+   * animation rules using them (animation rules using keyframes defined
+   * elsewhere need not to be modified to allow for globally defined keyframes)
+   *
+   * For example, we convert this css:
+   *
+   * .box {
+   *   animation: box-animation 1s forwards;
+   * }
+   *
+   * @keyframes box-animation {
+   *   to {
+   *     background-color: green;
+   *   }
+   * }
+   *
+   * to this:
+   *
+   * .box {
+   *   animation: scopeName_box-animation 1s forwards;
+   * }
+   *
+   * @keyframes scopeName_box-animation {
+   *   to {
+   *     background-color: green;
+   *   }
+   * }
+   *
+   **/
+  private _scopeLocalKeyframes(cssText: string, scopeSelector: string): string {
+    // This set is used to collect all keyframes names (before scoping) so that they can
+    // later be used to scope the animation rules.
+    const unscopedKeyframes = new Set<string>();
+
+    const COMMA_IN_PLACEHOLDER = '%COMMAIN%';
+
+    const str = (input?: string) => input || '';
+    const hasQuotes = (input: string) => input.includes('"') || input.includes('\'');
```

Comment:
This function could be moved to the top level and put at the bottom of this file.

Also I think that it would be slightly more performant and cleaner to use a regex. Currently if there is no `"` but the first character is `'`, then it would have to traverse the whole string looking for `"` before finding the`'` on the second pass.

```suggestion
function hasQuotes(input: string): boolean {
  return /['"]/.test(input);
}
```

Reply:
yeah I will make the change thanks a lot :slightly_smiling_face: 

I am no regex expert, so honestly I was not sure if two `includes` could be more performant than a single `regex` :sweat_smile: 

Also.... just out of curiosity (I really hope you don't mind my asking), why is declaring functions with `function` preferred here over using an expression with arrow functions and `const`? Is there any benefit?